### PR TITLE
Document the dependency on Python 2.7 and a C++ compiler

### PR DESCRIPTION
**Pull request by KevinSheedy**
As a newbie, here are a few changes to the README.md that would have saved me some time.

 - Documented the dependency on Python 2.7
 - Documented the dependency on a C++ compiler
 - Put the Install instructions before the code example (I tried running the code example before realising there were dependencies that weren't managed by npm install)
 - Removed the numbering from headings to make them easier to change

### CODE REVIEW

**Comment by cjbj** [comment]
Hi @KevinSheedy, thanks for the PR.  Before we can merge it, can you sign the OCA as referenced in the README section formerly known as 7, https://github.com/oracle/node-oracledb/blob/master/CHANGELOG.md ?

**Comment by KevinSheedy** [comment]
Thanks Christopher
I've amended the commit messages to include the signature as requested.
https://github.com/oracle/node-oracledb/pull/313

Regards,
Kevin

**Comment by cjbj** [comment]
@KevinSheedy contributors need to sign a separate document too. See https://github.com/oracle/node-oracledb/blob/master/CONTRIBUTING.md and http://www.oracle.com/technetwork/community/oca-486395.html

**Comment by KevinSheedy** [comment]
CLA sent.
It would be great is you could simplify this process. Contributing to most other open source projects is a lot easier:
http://www.infoworld.com/article/2608006/javascript/joyent-makes-it-easier-to-contribute-code-to-node-js.html

**Comment by cjbj** [comment]
Thanks @KevinSheedy

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # node-oracledb version 1.4
 
-## <a name="about"></a> 1. About node-oracledb
+## <a name="about"></a> About node-oracledb
```
Comment (cjbj) [inline README.md]: The section numbers are very useful when pointing people to doc, e.g. in a phone call.  I would keep them.

Comment (KevinSheedy) [inline README.md]: It's not a huge issue but in general, when I'm developing for a particular platform, I try to follow the conventions of that platform. This makes it easier for the community to understand. In this case, on the npm platform, if you look at the top 15 libraries on the [npm homepage](https://www.npmjs.com/) you can see that none of these 15 libraries number their bullet points. I think it's a markdown convention really. But like I said it's, only a small thing.

Comment (cjbj) [inline README.md]: Another point in your favor is that this is only a brief readme, and not so subject to the 'maze of twisty little passage' problem that can occur when trying to point people to sections of longer doc.

**Comment by cjbj** [comment]
Merged to 1.6. Thanks @KevinSheedy!

**Pull request closed without merging**